server/message_hooks_test.go

```diff
@@ -2880,3 +2881,136 @@ func TestUserWillLogin(t *testing.T) {
 		})
 	}
 }
+
+func TestMessageHasBeenPosted(t *testing.T) {
+	assert := require.New(t)
```

Comment:
I think this `assert` is tied to the wrong `testing.T` -- typically, you'd want it anchored to the nearest one, i.e. recreating it inside each `t.Run`.